qmi/instruments/digilent/analog_discovery.py

```diff
@@ -316,53 +328,59 @@ def prepare_analog_input_record(self,
             analog_in.channelFilterSet(channel, pydwf.DwfAnalogInFilter(adc_filter.value))
         except pydwf.DwfLibraryError as library_error:
             raise QMI_InstrumentException(str(library_error)) from library_error
+
         # Wait at least 2 seconds with Analog Discovery for the offset to stabilize, before the first reading after
         # device open or offset/range change
         time.sleep(2)
 
     @rpc_method
-    def get_analog_input_record(self, channel: int, amount_samples: Optional[int] = None) -> typing.List[float]:
-        """Obtain record of input voltages on specified channel.
+    def get_analog_input_record(self, channel: int, amount_samples: int | None = None) -> list[float]:
+        """Obtain a record of input voltages from specified channel.
 
-        Args:
-            channel: the desired channel to query.
+        Parameters:
+            channel:        The target channel to query.
             amount_samples: The amount of samples to take. Only relevant when record length is set to -1.
 
         Returns:
-            Record of input voltages.
+            sample_record:  A record of input voltages.
 
         Raises:
-            QMI_UsageException: Raised when a device is used before it is opened.
-            QMI_InstrumentException: QMI_InstrumentException is raised when data is corrupt or lost or when an
-                unexpected instrument error occurs.
+            QMI_UsageException:      Raised when a device is used before it is opened.
+            QMI_InstrumentException: Raised when data is corrupt or lost or when an unexpected instrument error occurs.
         """
         if self._device is None:
             raise QMI_UsageException("Device should be opened before use.")
 
         analog_in = self._device.analogIn
-        sample_record: np.ndarray = np.array([])
+        sample_record: np._typing.NDArray[np.float64] = np.array([])
```

Comment:
why are you using `np._typing` instead of just `numpy.typing`: https://numpy.org/devdocs/reference/typing.html#numpy.typing.NDArray

Reply:
Oh, my IDE said that there's no reference to 'typing' in '__init__.pyi'.  But it does work if I do remove the underscore. So, I'll change it.